sdk/keyvault/azure-security-keyvault/azure/security/keyvault/certificates/_client.py

```diff
@@ -0,0 +1,222 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import Any, Dict, Mapping, Optional
+from datetime import datetime
+
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.transport import RequestsTransport
+from azure.security.keyvault._internal import _BearerTokenCredentialPolicy
+
+from .._generated import KeyVaultClient
+from ._models import (
+    Certificate,
+    CertificateBase,
+    DeletedCertificate,
+    CertificatePolicy,
+    Issuer,
+    IssuerBase,
+    Contact,
+    CertificateOperation,
+)
+
+
+class CertificateClient:
+    @staticmethod
+    def create_config(**kwargs):
+        pass  # TODO
+
+    def __init__(self, vault_url, credentials, config=None, api_version=None, **kwargs):
+        if not credentials:
+            raise ValueError("credentials")
+
+        if not vault_url:
+            raise ValueError("vault_url")
+
+        self._vault_url = vault_url
+
+        if api_version is None:
+            api_version = KeyVaultClient.DEFAULT_API_VERSION
+
+        config = config or KeyVaultClient.get_configuration_class(api_version)(credentials)
+
+        # TODO generated default pipeline should be fine when token policy isn't necessary
+        policies = [
+            config.headers_policy,
+            config.user_agent_policy,
+            config.proxy_policy,
+            _BearerTokenCredentialPolicy(credentials),
+            config.redirect_policy,
+            config.retry_policy,
+            config.logging_policy,
+        ]
+        transport = RequestsTransport(config)
```

Comment:
The caller should be able to provide a transport instance. I'm fine with getting this in and  open an issue to track it...

Reply:
Similarly, all the sub-clients will inherit a class which handles this.